spec/views/users/index.html.erb_spec.rb

```diff
@@ -4,17 +4,29 @@
   before(:each) do
     @users = assign(:users, FactoryBot.create_list(:user, 3))
     sign_in @users.first
-
-    render
   end
 
   it "renders a list of users" do
+    render
     @users.each do |user|
       expect(rendered).to match user.email
     end
   end
 
-  it "has an + button to add a contact" do
+  it "has an + button to send contact request" do
+    render
     expect(rendered).to have_button('+', count: @users.length)
   end
+
+  it "does not have a + button for already requested user" do
+    @users.second.contact_requests << @users.first
+    render
+    expect(rendered).not_to have_css('form.button_to[action="/users/2/contact_requests"]')
+  end
+
+  it "does not have a + button for already existing contacts" do
+    @users.first.contacts << @users.second
+    render
+    expect(rendered).not_to have_css('form.button_to[action="/users/2/contact_requests"]')
```

Comment:
One could use the path helpers here (and in the test case above)  
`user_contact_requests GET    /users/:user_id/contact_requests(.:format)                                               contact_requests#index`

Reply:
the path helpers are not available in a view test as far as I tried and found stuff

Reply:
Ok, then it's fine :+1: